include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,330 @@
+      next_idx_ = next_idx; // update next behavior
+      if (!behaviors_[next_idx_]->is_initialized()) {
+        behaviors_[next_idx_]->Init();
+      }
+      behaviors_[selected_idx]->Stop(*behaviors_[next_idx_]); // stop current behavior
+      switching_behaviors_ = true;  // indicate transition is in progress
+    } else {
+      LOG_WARN("Invalid behavior selected: %d", next_idx);
+    }
+  }
+
+  /**
+   * @brief Set the current behavior
+   * @param name name of behavior in `behaviors_`
+   */
+  virtual void SetBehavior(const std::string &name) {
+    if (name == KILL_BEHAVIOR_NAME) { SetBehavior(KILL_ROBOT_IDX); }
+    else {
+      auto idx = get_behavior_index(name);
+      if (idx) { SetBehavior(idx.value()); }
+      else {
+        LOG_WARN("Invalid behavior selected: %s", name.c_str());
+      }
+    }
+  }
+
+  /**
+   * @brief Add a behavior to the behaviors list and initialize it
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   * @return index of behavior in behaviors list if added successfully,
+   * `std::nullopt` otherwise
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  std::optional<int> AddBehavior(ConstructorArgs &&... args) {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = ConstructBehavior<BehaviorType>(args...);
+    if (b) {
+      int idx = behaviors_.size();
+      std::string name = b->get_name();
+      names_.push_back(name);
+      name_to_idx_map_.emplace(std::make_pair(name, idx));
+      behaviors_.emplace_back(std::move(b));
+      return idx;
+    } else {
+      LOG_WARN(
+          "%s behavior could not be initialized and will not be added to behaviors list.",
+          b->get_name().c_str());
+      return {};
+    }
+  }
+
+  /**
+   * @brief Replaces the default behavior with a user-defined one
+   * @note This is included to provide support for custom default robot
+   *       behaviors. If this function is not called, the default behavior will
+   *       be a `kodlab::OffBehavior`.
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  void SetDefaultBehavior(ConstructorArgs &&... args) {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = ConstructBehavior<BehaviorType>(args...);
+    if (b) {
+      name_to_idx_map_.erase(default_name_);
+      default_name_ = b->get_name();
+      name_to_idx_map_.emplace(std::pair(default_name_, DEFAULT_IDX));
+      names_[0] = default_name_;
+      behaviors_[0] = std::move(b);
```

Comment:
shouldnt these use `DEFAULT_IDX`?

Reply:
Yes they should, fixed in 57a65848a1b429d69bd735eddc489b60878e210b